File: 프로그래머스/lv0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        int answer = 1, i=Math.min(a,b);
        while(i>1){
            if(a%i==0 && b%i==0){
                a/=i;
                b/=i;
            }
            i--;
        }
        while(b>1){
            if(b%2==0){
                b/=2;
            }else if(b%5==0){
                b/=5;  
            }else{
                answer = 2;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120878. 유한소수 판별하기/유한소수 판별하기.js
function solution(a, b) {
    var answer = 0;
    var b_prime = [];
    while(b>1){
        for(var i=2; i<=b; i++){
            if(b%i===0){
                b = Math.floor(b/i);
                b_prime.push(i);
                break;
            }
        }
    }
    b_prime.map((p, i) => {
        if(a%p===0){
            b_prime[i] = 2;
            a = Math.floor(a/p);
        }
    })
    return b_prime.join("").replace(/2|5/g, "")==="" ? 1 : 2;
}